### ISSUE TITLE
superfluous-parens triggered when it should not

### ISSUE THREAD

**Issue by aleb** [PyCQA/pylint#4962]
### Bug description

```python
def _(a):
    return a

for a, b in ((3, _(4)), (1, _(2))):
    print(a, b)
```

### Command used

```shell
$ pylint ~/a.py
```

### Pylint output

```shell
************* Module a
/home/aleb/a.py:4:0: C0325: Unnecessary parens after 'in' keyword (superfluous-parens)
```

### Expected behavior

I'd expect superfluous-parens not to be printed since the parenthesis are required.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.8.8 (default, Nov 10 2011, 15:00:00) 
[GCC 10.2.0]
```

**Comment by Pierre-Sassoulas** [comment]
This is supposed to be fixed in the main branch, and will be available in 2.11.0. Can you confirm by installing the latest pylint with : ``pip install git+git://github.com/PyCQA/pylint.git@main#egg=pylint -U``, please ?

**Comment by aleb** [comment]
Thanks! I believe you. We'll ignore this warning for now.
